rewriter/rewriter.cc

```diff
@@ -47,6 +48,7 @@ class Rewriterer {
         Prop::run(ctx, classDef);
         TypeMembers::run(ctx, classDef);
         Singleton::run(ctx, classDef);
+        Visibility::run(ctx, classDef);
```

Comment:
idk if you care about this, but

```ruby
private
attr_reader :foo
```

is not going to make `foo` private.

Reply:
The attr_reader DSL fires after your Visibility rewriter pass.

Essentially, what you want is for your rewriter pass to fire "after all other rewriter passes," because any rewriter pass might introduce new methods.

With that in mind, have you considered doing this in namer instead? Namer has a view of "all the methods that are defined" that rewriter passes don't (because namer comes after rewriter)

Reply:
Hmm... @bss What do you think about this? How do you all feel about shipping this as is and keeping that for a separate change?

Reply:
> How do you all feel about shipping this as is and keeping that for a separate change?

(I mention being fine with this in a comment below fwiw)

Reply:
We'll move the visibility rewriter pass as the last rewriter so it picks up any new methods that are added.